### PR TITLE
Updates for dynabench

### PR DESCRIPTION
**Pull request by neelsjain**
Changes 1) Updated the OUTPUT_FIELD from answer to label 2) changed how is_compliance is read and how the transscript and policies are read 3) added torch.no_grads() to a couple places

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -138,16 +138,16 @@ def is_valid_row(example):
...
         
     #########
     # 1. Get the policies and transcripts
     #########
     if is_compliance_dataset:
         # DynaGuard dataset:
-        transcripts = [extract_xml_answer(x[INPUT_FIELD], "<transcript>", "</transcript>") for x in dataset]
-        policies = [extract_xml_answer(x[INPUT_FIELD], "<rules>", "</rules>") for x in dataset]
+        transcripts = [x["transcript"] for x in dataset]
+        policies = [x["policy"] for x in dataset]
```
[the remark this answers is not in the document]
Comment (neelsjain) [reply eval.py:150]: @montehoover I don't think this change suggestion from copilot matters
